corehq/apps/consumer_user/tests/test_views.py

```diff
@@ -0,0 +1,398 @@
+        register_uri = register_url(1000)
+        response = self.client.get(register_uri)
+        self.assertEqual(response.status_code, 400)
+        self.assertTemplateNotUsed(response, 'signup.html')
+
+
+class LoginTestCase(TestCase):
+
+    def setUp(self):
+        super().setUp()
+        self.login_url = reverse('consumer_user:consumer_user_login')
+        self.homepage_url = reverse('consumer_user:consumer_user_homepage')
+        self.client = Client()
+
+    def tearDown(self):
+        ConsumerUserCaseRelationship.objects.all().delete()
+        ConsumerUserInvitation.objects.all().delete()
+        ConsumerUser.objects.all().delete()
+        User.objects.all().delete()
+        super().tearDown()
+
+    def test_login_get(self):
+        response = self.client.get(self.login_url)
+        self.assertEqual(response.status_code, 200)
+        self.assertTemplateUsed(response, 'p_login.html')
+
+    def test_login_post(self):
+        email = 'log@log.in'
+        password = 'password'
+        user = User.objects.create_user(username=email, email=email, password=password)
+        ConsumerUser.objects.create(user=user)
+        post_data = {
+            'auth-username': email,
+            'auth-password': password,
+            'consumer_user_login_view-current_step': 'auth',
+        }
+        response = self.client.post(self.login_url, post_data)
+        self.assertEqual(response.status_code, 302)
+        self.assertRedirects(response, self.homepage_url)
+
+    def test_login_accept_invitation(self):
+        email = 'log1@log1.in'
+        password = 'password'
+        user = User.objects.create_user(username=email, email=email, password=password)
+        ConsumerUser.objects.create(user=user)
+        invitation = ConsumerUserInvitation.objects.create(case_id='6', domain='1', invited_by='I',
+                                                           email=email)
+        post_data = {
+            'auth-username': email,
+            'auth-password': password,
+            'consumer_user_login_view-current_step': 'auth',
+        }
+        response = self.client.post(login_accept_url(invitation.id), post_data)
+        self.assertEqual(response.status_code, 302)
+        invitation.refresh_from_db()
+        self.assertTrue(invitation.accepted)
+        self.assertRedirects(response, self.homepage_url)
+
+
+class SignalTestCase(TestCase):
+
+    def setUp(self):
+        self.domain = 'consumer-invitation-test'
+        self.factory = CaseFactory(self.domain)
+
+    def tearDown(self):
+        FormProcessorTestUtils.delete_all_cases()
+        FormProcessorTestUtils.delete_all_xforms()
+        ConsumerUserCaseRelationship.objects.all().delete()
+        ConsumerUserInvitation.objects.all().delete()
+        ConsumerUser.objects.all().delete()
+        User.objects.all().delete()
+        super().tearDown()
+
+    @override_settings(TESTS_SHOULD_USE_SQL_BACKEND=True)
+    def test_method_send_email(self):
```

Comment:
This test seems to never complete. 
```
$ ./manage.py corehq.apps.consumer_user.tests.test_views:SignalTestCase.test_method_send_email
```
with an empty database, this just hangs (you can add `--reusedb=reset` to start fresh).